lightning/src/routing/gossip.rs

```diff
@@ -130,6 +133,28 @@ pub struct NetworkGraph<L: Deref> where L::Target: Logger {
 	// Lock order: channels -> nodes
 	channels: RwLock<BTreeMap<u64, ChannelInfo>>,
 	nodes: RwLock<BTreeMap<NodeId, NodeInfo>>,
+	// Lock order: removed_channels -> removed_nodes
+	//
+	// NOTE: In the following `removed_*` maps, we use seconds since UNIX epoch to track time instead
+	// of `std::time::Instant`s for a few reasons:
+	//   * We want it to be possible to do tracking in no-std environments where we can compare
+	//     a provided current UNIX timestamp with the time at which we started tracking.
+	//   * In the future, if we decide to persist these maps, they will already be serializable.
+	//   * Although we lose out on the platform's monotonic clock, the system clock in a std
+	//     environment should be practical over the time period we are considering (on the order of a
+	//     week).
+	//
+	/// Keeps track of short channel IDs for channels we have explicitly removed due to permanent
+	/// failure so that we don't resync them from gossip. Each SCID is mapped to the time in seconds
```

Comment:
nit: I think this comment is too verbose and may be compressed x2.

Keeps track of short channel IDs for channels we have explicitly removed due to permanent failure so that we don't resync them from gossip. Each SCID is mapped to the time (in seconds) it was removed so that once some time passes, we can potentially resync it from gossip again.

Reply:
Thanks, I've added these to the latest (squashed) changes :)